### ISSUE TITLE
[Bug]: pr-str through pr-db broken (not readable)

### ISSUE THREAD

**Issue by FalseProtagonist** [replikativ/datahike#722]
### What version of Datahike are you using?

xx

### What version of Java are you using?

xx

### What operating system are you using?

xx

### What database EDN configuration are you using?

xx

### Describe the bug

When you pr-str a DB or something like a TxReport containing a DB, the resulting str is not read-string-able because the pr-db implementation doesn't correctly escape quotes, because internally it uses str not pr-str. being read-string-able is the fundamental contract of pr-str.

Also with love this bug form is completely insane I'm not writing all this when I can see the bug in the source code on the latest branch.

### What is the expected behaviour?

xx

### How can the behaviour be reproduced?

xx

**Comment by whilo** [comment]
Hey @FalseProtagonist, thanks for opening the issue. I agree that it should be readable with read-string and this should be fixed. It looks like you have already identified a potential fix. Could you open a PR by any chance (ideally for all the issues you opened)? In case where tests will be failing I can help to fix them.

**Comment by harismh** [comment]
The ClojureScript port #748 switched `str` calls with `pr-str` calls: https://github.com/replikativ/datahike/blame/main/src/datahike/db.cljc#L688. So, I feel this bug is already fixed.

**Comment by whilo** [comment]
@harismh Yes, that is likely the case. In 2024 cljs was also not yet fully supported, only last summer I reactivated the port and made it also work durably. @FalseProtagonist if the issue persists, please reopen.
